### PR TITLE
Added types for audio-type

### PR DESCRIPTION
**Pull request by ctoth**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x]The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/audio-type/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  export function isOgg(buf: Uint8Array | ArrayBuffer | null): boolean;
+
+  export function isFlac(buf: Uint8Array | ArrayBuffer | null): boolean;
+
+  export function isM4a(buf: Uint8Array | ArrayBuffer | null): boolean;
+
+  export function isOpus(buf: Uint8Array | ArrayBuffer | null): boolean;
+
+  export function isQoa(buf: Uint8Array | ArrayBuffer | null): boolean;
+}
```
Comment (MysteryBlokHed) [inline types/audio-type/index.d.ts:17]: `declare module` isn’t necessary when the file is only for one module:

```suggestion
export default function audioType(buf: Uint8Array | ArrayBuffer | null): string | undefined;

export function isMp3(buf: Uint8Array | ArrayBuffer | null): boolean;

export function isWav(buf: Uint8Array | ArrayBuffer | null): boolean;

export function isOgg(buf: Uint8Array | ArrayBuffer | null): boolean;

export function isFlac(buf: Uint8Array | ArrayBuffer | null): boolean;

export function isM4a(buf: Uint8Array | ArrayBuffer | null): boolean;

export function isOpus(buf: Uint8Array | ArrayBuffer | null): boolean;

export function isQoa(buf: Uint8Array | ArrayBuffer | null): boolean;
```

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/audio-type/audio-type-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 1b5c64a2c30e3e28957a9da8317c96b0ce63c131
</p>

File: types/audio-type/package.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+    "private": true,
+    "name": "@types/audio-type",
+    "version": "2.2.9999",
```
Comment (andrewbranch) [inline types/audio-type/package.json:4]:
```suggestion
    "version": "2.2.9999",
    "type": "module",
```

must match the implementation package: https://unpkg.com/browse/audio-type@2.2.1/package.json

File: types/audio-type/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+export default function audioType(buf: Uint8Array | ArrayBuffer | null): string | undefined;
+
+export function isMp3(buf: Uint8Array | ArrayBuffer | null): boolean;
```
[the remark this answers is not in the document]
Comment (andrewbranch) [reply types/audio-type/index.d.ts:3]: I agree; the best thing is just to remove `null` from all these.

**Review by andrewbranch — CHANGES_REQUESTED** [review]

**Review by andrewbranch — APPROVED** [review]

File: types/audio-type/tslint.json [hunk 4]
```diff
@@ -0,0 +1 @@
+{ "extends": "@definitelytyped/dtslint/dt.json" }
```
Comment (jakebailey) [inline types/audio-type/tslint.json:1]: You should remove this file; it is no longer required.

**Review by jakebailey — CHANGES_REQUESTED** [review]

**Pull request closed without merging**